iothub_client/samples/pnp/pnp_temperature_controller/pnp_thermostat_component.c

```diff
@@ -0,0 +1,367 @@
+
+// Format string to create an ISO8601 time.  This corresponds to the DTDL datetime schema item.
+static const char g_ISO8601Format[] = "%04d-%02d-%02dT%02d:%02d:%02dZ";
+// Format string for sending temperature telemetry
+static const char g_temperatureTelemetryBodyFormat[] = "{\"temperature\":%.02f}";
+// Format string for building getMaxMinReport response
+static const char g_minMaxCommandResponseFormat[] = "{\"maxTemp\":%.2f,\"minTemp\":%.2f,\"avgTemp\":%.2f,\"startTime\":\"%s\",\"endTime\":\"%s\"}";
+// Format string for sending maxTempSinceLastReboot property
+static const char g_maxTempSinceLastRebootPropertyFormat[] = "%.2f";
+// Format of the body when responding to a targetTemperature 
+static const char g_targetTemperaturePropertyResponseFormat[] = "%.2f";
+
+// Start time of the program, stored in ISO8601 format string for UTC
+char g_programStartTime[128] = {0};
+
+
+// Response description is an optional, human readable message including more information
+// about the setting of the temperature.  Because we accept all temperature requests, we 
+// always indicate a success.  An actual sensor could optionally return information about
+// a temperature being out of range or a mechanical issue on the device on error cases.
+static const char g_temperaturePropertyResponseDescription[] = "success";
+
+//
+// PNP_THERMOSTAT_COMPONENT simulates a thermostat component
+// (as in thermostat1 or thermostat2 in the TemperatureController model).  We need separate data structures
+// because the components can be independently controlled.
+//
+typedef struct PNP_THERMOSTAT_COMPONENT_TAG
+{
+    // Name of this component
+    char componentName[PNP_MAXIMUM_COMPONENT_LENGTH + 1];
+    // Current temperature of this thermostat component
+    double currentTemperature;
+    // Minimum temperature this thermostat has been at during current execution run of this thermostat component
+    double minTemperature;
+    // Maximum temperature thermostat has been at during current execution run of this thermostat component
+    double maxTemperature;
+    // Number of times temperature has been updated, counting the initial setting as 1.  Used to determine average temperature of this thermostat component
+    int numTemperatureUpdates;
+    // Total of all temperature updates during current exceution run.  Used to determine average temperature of this thermostat component
+    double allTemperatures;
+}
+PNP_THERMOSTAT_COMPONENT;
+
+//
+// BuildUtcTimeFromCurrentTime writes the current time, in ISO 8601 format, into the specified buffer
+//
+static bool BuildUtcTimeFromCurrentTime(char* utcTimeBuffer, size_t utcTimeBufferSize)
+{
+    bool result;
+    time_t currentTime;
+    struct tm * currentTimeTm;
+
+    time(&currentTime);
+    currentTimeTm = gmtime(&currentTime);
+
+    if (snprintf(utcTimeBuffer, utcTimeBufferSize, g_ISO8601Format, currentTimeTm->tm_year + 1900, currentTimeTm->tm_mon, currentTimeTm->tm_mday, 
```

Comment:
any reason for not using `strftime`?

Reply:
Fixing and also in simple thermostat which had same sprintf() pattern.